Examples/src/main/kotlin/com/mapbox/vision/examples/POIView.kt

```diff
@@ -0,0 +1,167 @@
+package com.mapbox.vision.examples
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.Rect
+import android.os.Handler
+import android.util.AttributeSet
+import android.view.View
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.vision.VisionReplayManager
+import com.mapbox.vision.examples.POIActivityKt.POI
+import com.mapbox.vision.mobile.core.models.frame.ImageSize
+import com.mapbox.vision.mobile.core.models.position.GeoCoordinate
+import com.mapbox.vision.mobile.core.models.position.GeoLocation
+import com.mapbox.vision.mobile.core.models.world.WorldCoordinate
+import kotlin.math.max
+import kotlin.math.min
+
+class POIView
+@JvmOverloads
+constructor(
+    context: Context,
+    attrs: AttributeSet? = null
+) : View(context, attrs) {
+
+    companion object {
+
+        // POI will start to appear at this distance, starting with transparent and appearing gradually
+        private const val DRAW_LABEL_MIN_DISTANCE_METERS = 250
+
+        // POI will start to appear from transparent to non-transparent during this first meters of showing distance
+        private const val DISTANCE_FOR_ALPHA_APPEAR_METERS = 50
+        private const val LABEL_SIZE_METERS = 8
+        private const val LABEL_ABOVE_GROUND_METERS = 4
+    }
+
+    var poiList: List<POI> = arrayListOf()
+
+    private val paint = Paint()
+    private val uiHandler = Handler()
+    private var poiDrawDataList: MutableList<POIDrawData> = mutableListOf()
+
+    data class POIState(val poi: POI, val distanceToVehicle: Int, val worldCoordinate: WorldCoordinate?)
+
+    private data class POIDrawData(
```

Comment:
I'd move to separate file but up to you and remove `private`.

Reply:
extracted to separate package